internal/app/api/auth/auth.go

```diff
@@ -0,0 +1,138 @@
+		// Must return user data as user identifier, it will be stored in Claim Array. Required.
+		// Check error (e) to determine the appropriate error message.
+		Authenticator: func(ctx *gin.Context) (interface{}, error) {
+			payload := &generated.UserLogin{}
+
+			if ctx.ContentType() == "application/json" {
+				if err := ctx.ShouldBind(payload); err != nil {
+					return nil, err
+				}
+			} else if ctx.ContentType() == "application/x-www-form-urlencoded" {
+				payload.Username = ctx.PostForm("username")
+				payload.Password = ctx.PostForm("password")
+			} else {
+				errors.AbortWithError(http.StatusBadRequest, "content-type not supported", ctx)
+			}
+
+			user := &models.User{}
+			result := db.First(user, "email = ?", strings.ToLower(payload.Username))
+			if result.Error != nil {
+				log.Err(result.Error).Msg("failed to authenticate user - database query failed")
+				return nil, jwt.ErrFailedAuthentication
+			}
+
+			if err := utils.VerifyPassword(user.Password, payload.Password); err != nil {
+				log.Err(jwt.ErrFailedAuthentication).Str("userId", user.ID.String()).Msg("invalid password")
+				return nil, jwt.ErrFailedAuthentication
+			}
+
+			return user, nil
+		},
+		Unauthorized: func(c *gin.Context, code int, message string) {
+			errors.AbortWithError(code, message, c)
+			return
+		},
+		Authorizator: func(data interface{}, c *gin.Context) bool {
+			if _, ok := c.Get(generated.PinmanAuthScopes); ok {
+				if _, ok := data.(*models.User); ok {
+					//TODO: Validate authorization
+				}
+			}
+
+			return true
+		},
+		LoginResponse: func(c *gin.Context, code int, token string, expire time.Time) {
+			c.JSON(http.StatusOK, generated.TokenResponse{
+				AccessToken: token,
+				Expire:      expire.Format(time.RFC3339),
+			})
+		},
+		RefreshResponse: func(c *gin.Context, code int, token string, expire time.Time) {
+			c.JSON(http.StatusOK, generated.TokenResponse{
+				AccessToken: token,
+				Expire:      expire.Format(time.RFC3339),
+			})
+		},
+		// TokenLookup is a string in the form of "<source>:<name>" that is used
+		// to extract token from the request.
+		// Optional. Default value "header:Authorization".
+		// Possible values:
+		// - "header:<name>"
+		// - "query:<name>"
+		// - "cookie:<name>"
+		// - "param:<name>"
+		TokenLookup: "header:Authorization",
+		// TokenLookup: "query:token",
+		// TokenLookup: "cookie:token",
+
+		// TokenHeadName is a string in the header. Default value is "Bearer"
+		TokenHeadName: "Bearer",
+
+		// TimeFunc provides the current time. You can override it to use another time value. This is useful for testing or if your server uses a different time zone than your tokens.
+		TimeFunc: time.Now,
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create middleware: %v", err)
+	}
+
+	return authMiddleware, nil
+}
+
+func GetUser(ctx *gin.Context) *models.User {
+	return ctx.MustGet(IdentityKey).(*models.User)
```

Comment:
Yep good point, in theory this should never happen because the ctx user only get set after the request is authenticated, but still good to be a bit more verbose.